### PR TITLE
Fix Issue with SAML Document for IDP

### PR DESCRIPTION
**Pull request by r351574nc3**
## what
* Switching from `join` to list notation  when choosing default for `saml_provider_arn`

## why
* join results in empty string which forces `try` block to succeed on first try each time
    * the try block is broken and never really works

## references
N/A

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -3,7 +3,7 @@ locals {
   certificate_backends       = ["ACM", "SSM"]
   mutual_enabled             = var.authentication_type == "certificate-authentication"
   federated_enabled          = var.authentication_type == "federated-authentication"
-  saml_provider_arn          = local.federated_enabled ? try(join("", aws_iam_saml_provider.default.*.arn), var.saml_provider_arn) : null
+  saml_provider_arn          = local.federated_enabled ? try(aws_iam_saml_provider.default[0].arn, var.saml_provider_arn) : null
```
Comment (nitrocode) [inline main.tf:6]: Why this change ?

File: main.tf [hunk 2]
```diff
@@ -132,7 +132,7 @@ module "cloudwatch_log" {
 resource "aws_iam_saml_provider" "default" {
   count = var.saml_metadata_document != null ? 1 : 0
 
-  name                   = var.name
+  name                   = module.this.id
```
Comment (nitrocode) [inline main.tf:135]: Would changing the name cause a recreation of this resource ?

File: main.tf (hunk 1 above)
Comment (nitrocode) [reply main.tf:6]: Instead of using a `try` here, can we use a `coalesce` instead ?

```suggestion
  saml_provider_arn          = local.federated_enabled ? coalesce(join("", aws_iam_saml_provider.default.*.arn), var.saml_provider_arn)
```

or would that return the same error ?

File: main.tf (hunk 2 above)
Comment (r351574nc3) [reply main.tf:135]: Yes. Changing the name of the resource would result in a creation.

**Review by nitrocode — APPROVED** [review]
@r351574nc3 says it was tested so I'll approve this. LGTM
